core/views/browse.py

```diff
@@ -196,6 +205,23 @@ def page(request, lccn, date, edition, sequence):
         else:
             explanation = ""
 
+    # if no word highlights were requests, see if the user came
+    # from search engine results and attempt to highlight words from their
+    # query by redirecting to a url that has the highlighted words in it
+    if not words:
+        try:
+            words = _search_engine_words(request)
+            words = '+'.join(words)
+            if len(words) > 0:
+                url = '%s?%s#%s' % (urlresolvers.reverse('chronam_page_words', kwargs=path_parts), request.GET.urlencode(), words)
+                return HttpResponseRedirect(url)
+        except Exception, e:
```

Comment:
Since we're touching this code, this should be upgraded to the Python 2.5+ `except Exception as e:` syntax since that's now a syntax error on Python 3.